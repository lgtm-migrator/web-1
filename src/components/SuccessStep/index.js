import React from "react"
import AlertTitle from "@material-ui/lab/AlertTitle"
import Alert from "@material-ui/lab/Alert"
import { Article, H2, Paragraph } from "muy"
import CardContent from "@material-ui/core/CardContent"
import Card from "@material-ui/core/Card"
import Fade from "@material-ui/core/Fade"
import Divider from "@material-ui/core/Divider"
import ExpansionPanel from "@material-ui/core/ExpansionPanel"
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary"
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails"
import Button from "@material-ui/core/Button"

const SuccessStep = ({ carPlate, carMake, carMakeModel, onClick }) => (
  <Article>
    <Fade in={true} timeout={0}>
      <Alert>
        <AlertTitle>Recebemos a sua denúncia anônima com sucesso!</AlertTitle>
      </Alert>
    </Fade>
    <Divider />
    <Fade in={true} timeout={1000}>
      <Card>
        <CardContent>
          <H2 gutterBottom={true}>E agora?</H2>
          <Paragraph>
            Iremos analisar o conteúdo da sua denúncia antes de publicá-la.
          </Paragraph>
          <Paragraph variant={"body2"}>
            Se estiver tudo certo, publicaremos sua denúncia em nosso site, e
            depois encaminharemos as informações para o órgão responsável pela
            fiscalização de trânsito da sua cidade.
          </Paragraph>
        </CardContent>
      </Card>
    </Fade>
    <Divider />
    <ExpansionPanel>
      <ExpansionPanelSummary>
        Revisar dados da sua última denúncia
      </ExpansionPanelSummary>
      <ExpansionPanelDetails>
        <dl>
          <dt>Placa</dt>
          <dd>{carPlate}</dd>
          <dt>Marca</dt>
          <dd>{carMake}</dd>
          <dt>Modelo</dt>
          <dd>{carMakeModel} (ou similar)</dd>
        </dl>
      </ExpansionPanelDetails>
    </ExpansionPanel>
    <Divider />
    <Button color={"primary"} variant={"contained"} onClick={onClick}>
      Fazer outra denúncia
    </Button>
  </Article>
)

export default SuccessStep
