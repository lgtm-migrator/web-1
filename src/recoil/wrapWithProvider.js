import React from "react"
import { RecoilRoot } from "recoil"

// eslint-disable-next-line react/display-name,react/prop-types
export const wrapWithProvider = ({ element }) => (
  <RecoilRoot>{element}</RecoilRoot>
)
