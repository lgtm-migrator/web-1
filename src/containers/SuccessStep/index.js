import React from "react"
import SuccessStep from "../../components/SuccessStep"

import { useSetActiveStepState } from "../../hooks/useStepsNavigation/useActiveStepState"
import { useParkingReportState } from "../../hooks/useParkingReportState"

export default () => {
  const setActiveStepIndex = useSetActiveStepState()
  const [parkingReportState, setParkingReportState] = useParkingReportState()
  const { carPlate, carMake, carMakeModel } = parkingReportState

  const onClick = () => {
    setActiveStepIndex(0)
    setParkingReportState({
      uuid: null,
      carFrontPhotoPreviewUrl: null,
      carPlate: "",
      currentPosition: null,
      isCarFrontPhotoValid: true,
      isLocationValid: true,
    })
  }

  const props = {
    carPlate,
    carMake,
    carMakeModel,
    onClick,
  }
  return <SuccessStep {...props} />
}
