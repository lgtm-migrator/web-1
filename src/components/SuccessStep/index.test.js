import React from "react"
import renderer from "react-test-renderer"
import { render, screen, createEvent, fireEvent } from "@testing-library/react"
import SuccessStep from "."
import { RecoilRoot } from "recoil"

jest.mock("@material-ui/core/fade")

describe("SuccessStep", () => {
  const mockOnClick = jest.fn()
  const props = {
    onClick: mockOnClick,
  }

  it("should render correctly", () => {
    const tree = renderer
      .create(
        <RecoilRoot>
          <SuccessStep {...props} />
        </RecoilRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render 'Fazer outra denúncia' button", () => {
    render(
      <RecoilRoot>
        <SuccessStep {...props} />
      </RecoilRoot>
    )
    const element = screen.getByText("Fazer outra denúncia", {
      selector: "button span",
    })
    expect(element).toBeTruthy()
  })

  it("should call onClick", () => {
    render(
      <RecoilRoot>
        <SuccessStep {...props} />
      </RecoilRoot>
    )

    const element = screen.getByText("Fazer outra denúncia", {
      selector: "button span",
    })

    fireEvent.click(element)

    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
})
