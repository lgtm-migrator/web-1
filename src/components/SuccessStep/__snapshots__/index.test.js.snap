// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessStep should render correctly 1`] = `
<article
  className="MuiBox-root MuiBox-root-2"
>
  <hr
    className="MuiDivider-root"
  />
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        Revisar dados da sua última denúncia
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <dl>
                <dt>
                  Placa
                </dt>
                <dd>
                  ABC1234
                </dd>
                <dt>
                  Marca
                </dt>
                <dd>
                  Kia
                </dd>
                <dt>
                  Modelo
                </dt>
                <dd>
                  Cerato (ou similar)
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Fazer outra denúncia
    </span>
  </button>
</article>
`;
